internal/integration/publish_test.go

```diff
@@ -120,25 +129,115 @@ func TestPublish(t *testing.T) {
+	wait(t, exportPeriod+500*time.Millisecond, "Waiting before export batches")
+	server.ExportBatches(t)
+
+	wait(t, 500*time.Millisecond, "Waiting before staring workers")
+	server.StartExportWorkers(t)
+
+	memory, ok := env.Blobstore().(*storage.Memory)
+	if !ok {
+		t.Fatalf("can't use %t blobstore for verification", env.Blobstore())
+	}
+	keyExport := getKeysFromLatestBatch(t, "my-bucket", ctx, env, memory)
+
+	got := keyExport
+
+	wantedKeysMap := make(map[string]export.TemporaryExposureKey)
+	for _, key := range payload.Keys {
+		wantedKeysMap[key.Key] = export.TemporaryExposureKey{
+			KeyData:                    util.DecodeKey(key.Key),
+			TransmissionRiskLevel:      proto.Int32(int32(key.TransmissionRisk)),
+			RollingStartIntervalNumber: proto.Int32(key.IntervalNumber),
+		}
+	}
+
+	want := export.TemporaryExposureKeyExport{
+		StartTimestamp: nil,
+		EndTimestamp:   nil,
+		Region:         proto.String("TEST"),
+		BatchNum:       proto.Int32(1),
+		BatchSize:      proto.Int32(1),
+		SignatureInfos: nil,
+		Keys:           nil,
+	}
+
+	options := []cmp.Option{
+		cmpopts.IgnoreFields(want, "StartTimestamp"),
+		cmpopts.IgnoreFields(want, "EndTimestamp"),
+		cmpopts.IgnoreFields(want, "SignatureInfos"),
+		cmpopts.IgnoreFields(want, "Keys"),
+		cmpopts.IgnoreUnexported(want),
+	}
+
+	diff := cmp.Diff(got, &want, options...)
+	if diff != "" {
+		t.Errorf("%v", diff)
+	}
+
+	for _, key := range got.Keys {
+		s := util.ToBase64(key.KeyData)
+		wantedKey := wantedKeysMap[s]
+		gotKey := *key
+		diff := cmp.Diff(wantedKey, gotKey, cmpopts.IgnoreUnexported(gotKey))
+		if diff != "" {
+
+			t.Logf("WANT: %v", proto.MarshalTextString(&wantedKey))
+			t.Logf(" GOT: %v", proto.MarshalTextString(&gotKey))
+
+			t.Errorf("invalid key value: %v:%v", s, diff)
+		}
+	}
+
+	bytes, err := json.MarshalIndent(got, "", "")
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("can't marshal json results: %v", err)
 	}
-	defer resp.Body.Close()
 
-	resp, err = client.Get("/export/do-work")
+	t.Logf("%v", string(bytes))
+	// TODO: verify signature
+}
+
+func getKeysFromLatestBatch(t *testing.T, exportDir string, ctx context.Context, env *serverenv.ServerEnv, memory *storage.Memory) *export.TemporaryExposureKeyExport {
+	exportFile := getLatestFile(t, memory, ctx, exportDir)
+
+	t.Logf("Reading keys data from: %v", exportFile)
+
+	blob, err := env.Blobstore().GetObject(ctx, "", exportFile)
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer resp.Body.Close()
 
-	// TODO: verify export has the correct file
-	b, err := env.Blobstore().GetObject(ctx, "my-bucket", "index.txt")
+	keyExport, err := exportapi.UnmarshalExportFile(blob)
 	if err != nil {
-		t.Fatal(err)
+		t.Fatalf("can't extract export data: %v", err)
 	}
-	_ = b
 
-	// TODO: verify signature
+	return keyExport
+}
+
+func getLatestFile(t *testing.T, blobstore *storage.Memory, ctx context.Context, exportDir string) string {
+	files := blobstore.ListObjects(ctx, exportDir)
```

Comment:
Why don't we know? Shouldn't the export be name be constructed from information we know (/cc @gurayAlsac)?

Reply:
In theory the name is deterministic but, yeah, I suppose it depends on how long the test takes to execute and what the truncate window/batch period is whether we can directly construct the right name. It doesn't seem completely unreasonable (esp now w/ the debug endpoint) to have this method live in the non-test api, But I haven't dove into the full PR though to know how awkward that would make things.

Reply:
@sethvargo WDYT?

Reply:
Done.

Reply:
As long as we're listing objects from blobstore, I suppose no harm reading the index.txt file for the test (as Mike suggests) since that is in effect the same and also how clients would be doing it.